packages/webpack-cli/lib/webpack-cli.js

```diff
@@ -1691,6 +1703,35 @@ class WebpackCLI {
             }
         }
 
+        let extendedConfigsOptions = [];
+
+        // extend configurations provided in webpack configuration file
+        if (config.options.extends) {
+            extendedConfigsOptions.push(config.options);
+        } else if (Array.isArray(config.options)) {
+            extendedConfigsOptions = config.options.filter((config) => config.extends);
+        }
+
+        // extend configurations provided via --extends
+        if (options.extends) {
+            extendedConfigsOptions.push({ extends: options.extends });
+            delete options.extends;
+        }
+
+        extendedConfigsOptions.forEach(async (configOptions) => {
+            if (configOptions.extends) {
+                const newOptions = { config: [] };
+
+                configOptions.extends.map(async (value) => {
+                    newOptions.config.push(path.resolve(value));
+                });
+
+                delete configOptions.extends;
+
+                await this.resolveConfig(newOptions);
+            }
+        });
```

Comment:
We do not use `webpack-merge` logic here? it is working good

Reply:
We will also need to load the extended configurations, here we have only `paths` for extended configurations and not their `options`. 

Or am I missing something here?

Reply:
Ideally `--extends` means `--config=webpack.config.js --config=webpack.other.config.js --merge`, so I think we already have logic for this, for the next major release we will change logic for multiple `--config` https://github.com/webpack/webpack-cli/blob/master/packages/webpack-cli/lib/webpack-cli.js#L1619, I think we will throw an error on two and more config options without `--merge`, but we can reuse this logic for `--extends`, `--extends` like syntax sugar, also do not forget `--extends` in CLI should override `extends` on `webpack.config.js` (I think eslint works the same, but need check)